catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorFunctionListTest.java

```diff
@@ -0,0 +1,150 @@
+				{R.string.formula_editor_function_arcsin, R.string.formula_editor_function_arcsin_parameter, "arcsin"},
+				{R.string.formula_editor_function_arccos, R.string.formula_editor_function_arccos_parameter, "arccos"},
+				{R.string.formula_editor_function_exp, R.string.formula_editor_function_exp_parameter, "exp"},
+				{R.string.formula_editor_function_arctan, R.string.formula_editor_function_arctan_parameter, "arctan"},
+				{R.string.formula_editor_function_floor, R.string.formula_editor_function_floor_parameter, "floor"},
+				{R.string.formula_editor_function_ceil, R.string.formula_editor_function_ceil_parameter, "ceil"},
+				{R.string.formula_editor_function_max, R.string.formula_editor_function_max_parameter, "max"},
+				{R.string.formula_editor_function_min, R.string.formula_editor_function_min_parameter, "min"},
+				{R.string.formula_editor_function_length, R.string.formula_editor_function_length_parameter, "length"},
+				{R.string.formula_editor_function_letter, R.string.formula_editor_function_letter_parameter,
+						"letter"},
+				{R.string.formula_editor_function_join, R.string.formula_editor_function_join_parameter, "join"},
+				{R.string.formula_editor_function_number_of_items, R.string
+						.formula_editor_function_number_of_items_parameter, "number of items"},
+				{R.string.formula_editor_function_list_item, R.string.formula_editor_function_list_item_parameter,
+						"list item"},
+				{R.string.formula_editor_function_contains, R.string.formula_editor_function_contains_parameter,
+						"contains"}
+		});
+	}
+
+	@Parameter
+	public @StringRes int formulaEditorFunction;
+
+	@Parameter(1)
+	public @StringRes int formulaEditorFunctionParameter;
+
+	@Parameter(2)
+	public String testName;
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("FormulaEditorFunctionListTest");
+		script.addBrick(new ChangeSizeByNBrick(0));
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testFunctionsListElements() {
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_change_size_by);
+		onBrickAtPosition(1).onChildView(withId(R.id.brick_change_size_by_edit_text)).perform(click());
+
+		onView(withText(R.string.formula_editor_functions)).perform(click());
+
+		String formulaEditorFunctionString = UiTestUtils.getResourcesString(formulaEditorFunction);
+		String formulaEditorFunctionParameterString = UiTestUtils.getResourcesString(formulaEditorFunctionParameter);
+
+		String editorFunction = formulaEditorFunctionString + formulaEditorFunctionParameterString;
+		String selectedFunctionString = getSelectedFunctionString(editorFunction);
+
+		onData(allOf(is(instanceOf(String.class)), is(editorFunction)))
+				.inAdapterView(FormulaEditorFunctionListMatchers.isFunctionListView())
+				.check(matches(isDisplayed()));
+
+		onView(withText(editorFunction)).perform(click());
```

Comment:
dont use onView for clicking on this, you rely on call above onData to have scrolled you to the right place....

go with this instead, if you click on it, it also checks that it is there and visible (if not it will not be able to click on it...)

`onData(allOf(is(instanceOf(String.class)), is(editorFunction)))
				.inAdapterView(FormulaEditorFunctionListMatchers.isFunctionListView())
				.perform(click());`